### PR TITLE
Enable Central Package Management

### LINKED ISSUE #3434: [Feature Request] Move to use Central Package Manager
https://docs.microsoft.com/en-us/nuget/consume-packages/central-package-management

This will make handling package dependencies simpler.

### PR DESCRIPTION
**Pull request by pmaytak**
Fixes #3434

**Changes proposed in this request**
- Enabled Central Package Management for main project and tests.
- Disabled it for dev apps and cache compat tests.
- The package versions are in Directory.Packages.props.
   - They are hierarchical, so the one in tests folder overwrites the root one.

**Notes**
Currently Central Package Management doesn't support floating versions, but it will be in the next NuGet release: https://github.com/NuGet/NuGet.Client/pull/5440.

### CODE REVIEW

**Review by bgavrilMS — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 1]
```diff
@@ -242,32 +242,32 @@
...
-    <PackageReference Include="System.Security.SecureString" Version="4.3.0" />
+    <PackageReference Include="System.Security.SecureString" />
   </ItemGroup>
   <ItemGroup Condition=" '$(TargetFramework)' == '$(TargetFrameworkAndroid9)' ">
-    <PackageReference Include="Xamarin.Android.Support.CustomTabs" Version="28.0.0.3" />
+    <PackageReference Include="Xamarin.Android.Support.CustomTabs" />
   </ItemGroup>
   <ItemGroup Condition=" '$(TargetFramework)' == '$(TargetFrameworkAndroid10)'">
-    <PackageReference Include="Xamarin.AndroidX.Browser" Version="1.0.0" />
+    <PackageReference Include="Xamarin.AndroidX.Browser" VersionOverride="1.0.0" />
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:251]: maybe leave a comment here on why the version was overridden? that way we know the `TargetFrameworkNet6Android` version comes from package props and it uses the latest version

File: tests/CacheCompat/Directory.Packages.props [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)..\'))" />
```
Comment (pmaytak) [inline tests/CacheCompat/Directory.Packages.props:2]:
```suggestion
  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Packages.props', '$(MSBuildThisFileDirectory)..\'))" />
```

File: tests/devapps/Directory.Packages.props [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)..\'))" />
```
Comment (pmaytak) [inline tests/devapps/Directory.Packages.props:2]:
```suggestion
  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Packages.props', '$(MSBuildThisFileDirectory)..\'))" />
```

**Review by trwalke — APPROVED** [review]

File: Directory.Build.props [hunk 4]
```diff
@@ -7,30 +7,25 @@
...
-  
+
   <ItemGroup>
     <!-- PrivateAssets="All" is on these packages to ensure they're development dependencies
          and aren't included as transitionary dependencies to package consumers.
          For clarity, without PrivateAssets marked here, anyone consuming Microsoft.Identity.Client
          would also be forced to install these dependencies.  PrivateAssets avoids this problem. -->
-    <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="17.*" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="*" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" PrivateAssets="All" />
```
[the remark this answers is not in the document]
Comment (pmaytak) [reply Directory.Build.props:28]: Package.props has the `PackageVersion`, everywhere else has the `PackageReference`.

**Comment by pmaytak** [comment]
The [doc](https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management) says this feature is available on .NET SDK 6.0.300 but this does seem to work for any other platforms also. The builds look the same and I compared the packages which reference the same dependencies.  If you think we still should merge this in, please approve again.

**Review by bgavrilMS — APPROVED** [review]

**Review by gladjohn — APPROVED** [review]

**Pull request merged**